packages/commonwealth/server/routes/viewUserActivity.ts

```diff
@@ -18,34 +18,34 @@ export default async (
   const { id } = req.user;
 
   const query = `
-    SELECT nt.thread_id, nts.created_at as last_activity, nts.notification_data, nts.category_id,
-      MAX(thr.view_count) as view_count,
-      COUNT(DISTINCT oc.id) AS comment_count,
-      COUNT(DISTINCT tr.id) + COUNT(DISTINCT cr.id) AS reaction_count
-    FROM
-      (SELECT nnn.mx_not_id, nnn.thread_id,
-          ROW_NUMBER() OVER (ORDER BY mx_not_id DESC) as thread_rank
-        FROM
-        (SELECT DISTINCT nn.thread_id, nn.mx_not_id
-          FROM (SELECT (n.notification_data::jsonb->>'thread_id') AS thread_id,
-                  MAX(n.id) OVER (PARTITION BY (n.notification_data::jsonb->>'thread_id')) AS mx_not_id
-                FROM "Notifications" n
-                WHERE n.category_id IN('new-thread-creation','new-comment-creation')
-                  AND n.chain_id IN(SELECT a."chain" FROM "Addresses" a WHERE a.user_id = ?)
-                ORDER BY id DESC
-                FETCH FIRST 500 ROWS ONLY
-                ) nn
-          ) nnn
-      ) nt
-    INNER JOIN "Notifications" nts ON nt.mx_not_id = nts.id
-    LEFT JOIN "Comments" oc ON nt.thread_id = CAST(oc.thread_id AS VARCHAR)
-      --TODO: eval execution path with alternate aggregations
-    LEFT JOIN "Reactions" tr ON nt.thread_id = CAST(tr.thread_id AS VARCHAR)
-    LEFT JOIN "Reactions" cr ON oc.id = cr.comment_id
-    LEFT JOIN "Threads" thr ON thr.id = CAST(nt.thread_id AS int)
-    WHERE nt.thread_rank <= 50
-    GROUP BY nt.thread_id, nts.created_at, nts.notification_data, nts.category_id
-    ORDER BY nts.created_at DESC;
+      SELECT nt.thread_id, nts.created_at as last_activity, nts.notification_data, nts.category_id,
+             MAX(thr.view_count) as view_count,
+             COUNT(DISTINCT oc.id) AS comment_count,
+             COUNT(DISTINCT tr.id) + COUNT(DISTINCT cr.id) AS reaction_count
+      FROM
+          (SELECT nnn.mx_not_id, nnn.thread_id,
+                  ROW_NUMBER() OVER (ORDER BY mx_not_id DESC) as thread_rank
+           FROM
+               (SELECT DISTINCT nn.thread_id, nn.mx_not_id
+                FROM (SELECT n.thread_id AS thread_id,
```

Comment:
this inner select can be changed to:
```
SELECT n.thread_id AS thread_id,
                             MAX(n.id) OVER (PARTITION BY n.thread_id) AS mx_not_id
                      FROM "Notifications" n
                      WHERE n.category_id IN('new-thread-creation','new-comment-creation')
                        AND n.chain_id IN(SELECT a."chain" FROM "Addresses" a WHERE a.user_id = ?)
                      ORDER BY id DESC
                          FETCH FIRST 500 ROWS ONLY
```

to 

```
SELECT n.thread_id AS thread_id,
                             MAX(n.id) as mx_not_id
                      FROM "Notifications" n
                      WHERE n.category_id IN('new-thread-creation','new-comment-creation')
                        AND n.chain_id IN(SELECT a."chain" FROM "Addresses" a WHERE a.user_id = ?)
GROUP BY n.thread_id
```

Reply:
Including this in the next commit + adding documentation. I think it's important this becomes readable + understandable considering that it is technically our content recommendation query.

Reply:
@NakulManchanda lmk what you think of it now

Reply:
@timolegros 
Let's try this once if we can, uf new query is equivalent in performance
Let's add issue for this if we would like to refactor later

**my bad - refactor not just readability, its just weird SQL which we should replace!**